### ISSUE TITLE
Braille styled plot is all dotted when you paste it to any online pastebin service

### ISSUE THREAD

**Issue by Nakilon** [red-data-tools/unicode_plot.rb#59]
First I tried to paste the dot plot to https://gist.github.com/ but the background was all dotted.
Not clean like I see it in Terminal:

![image](https://user-images.githubusercontent.com/2870363/140195675-8c75e19a-96a1-4f2d-b1f1-1fea4f566e80.png)

Then I tried another pastebin service but it looks the same:

![image](https://user-images.githubusercontent.com/2870363/140195388-b625b5fa-ad4d-440d-93b3-8920f48cfc9f.png)

Is there any way to make it look clean?

P.S.: same result when using the `aha` (ansi to html converter) utility.

**Comment by kojix2** [comment]
Hi! @Nakilon

I was not able to reproduce this in my environment. But I can guess why it happened.

```ruby
irb(main):007:0> UnicodePlot.lineplot!(plot, x, y_cos, name: "cos(x)")
irb(main):008:0> plot.render
      ┌────────────────────────────────────────┐       
    1 │⠉⠑⢄⢀⠔⠉⠉⠙⢄⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⠜⠉⠉⠑⢄⢀⠔⠉⠉⠉⢢⠀⠀⠀⠀⠀⠀│ sin(x)
      │⠀⠀⢈⢎⠀⠀⠀⠀⠀⢣⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢠⠃⠀⠀⠀⠀⢈⢎⠀⠀⠀⠀⠀⠣⡀⠀⠀⠀⠀│ cos(x)
      │⠀⢠⠋⠘⡄⠀⠀⠀⠀⠀⢣⠀⠀⠀⠀⠀⠀⠀⠀⠀⢠⠃⠀⠀⠀⠀⢀⠎⠈⢆⠀⠀⠀⠀⠀⢣⠀⠀⠀⠀│       
      │⠀⡇⠀⠀⠘⡆⠀⠀⠀⠀⠈⢢⠀⠀⠀⠀⠀⠀⠀⢀⠇⠀⠀⠀⠀⠀⡎⠀⠀⠈⢆⠀⠀⠀⠀⠀⢣⠀⠀⠀│       
      │⡜⠀⠀⠀⠀⠘⡄⠀⠀⠀⠀⠈⢆⠀⠀⠀⠀⠀⠀⡎⠀⠀⠀⠀⠀⡸⠀⠀⠀⠀⠘⡄⠀⠀⠀⠀⠀⢇⠀⠀│       
      │⠉⠉⠉⠉⠉⠉⢫⠉⠉⠉⠉⠉⠙⡍⠉⠉⠉⠉⡝⠉⠉⠉⠉⠉⡹⠉⠉⠉⠉⠉⠉⠙⡍⠉⠉⠉⠉⠉⠉⠉│       
      │⠀⠀⠀⠀⠀⠀⠀⢣⠀⠀⠀⠀⠀⠸⡀⠀⠀⡼⠀⠀⠀⠀⠀⢠⠃⠀⠀⠀⠀⠀⠀⠀⢱⠀⠀⠀⠀⠀⠀⠀│       
      │⠀⠀⠀⠀⠀⠀⠀⠈⢆⠀⠀⠀⠀⠀⢣⡀⡰⠁⠀⠀⠀⠀⢠⠃⠀⠀⠀⠀⠀⠀⠀⠀⠀⢣⠀⠀⠀⠀⠀⠀│       
      │⠀⠀⠀⠀⠀⠀⠀⠀⠀⢣⠀⠀⠀⠀⠀⡱⡁⠀⠀⠀⠀⢠⠃⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠣⡀⠀⠀⠀⠀│       
   -1 │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠣⣀⣀⣀⠔⠁⠑⢄⣀⣀⡰⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠑⣄⡀⠀⠀│       
      └────────────────────────────────────────┘       
      0                                       10
=> nil
irb(main):009:0> "⠀".ord # copy & paste space from dot plot
=> 10240
irb(main):010:0> " ".ord # space
=> 32
```

See the character code. 
The spaces in the dot plot are not normal spaces, but 0x2800 - Braille Pattern Blank.
https://unicode-table.com/en/2800/
![image](https://user-images.githubusercontent.com/5798442/140232953-5ad76f61-3ef7-46be-b26f-c5f7f20fa358.png)

Perhaps you are using a programming-friendly font with visible Braille spaces to distinguish between normal spaces and special characters.

Sorry if I'm wrong.

By the way, is 3HD7.pdb a protein?
https://www.rcsb.org/structure/3hd7

**Comment by Nakilon** [comment]
Not sure what you mean by "can't reproduce". I see dots in your comment

![image](https://user-images.githubusercontent.com/2870363/140238568-139fd59d-5329-44b5-9c25-b219c9818efb.png)

while there are no dots in terminal screenshot in repo README. Yeah, macOS Terminal uses some good font that is probably not used by default either by Chrome or by pastebin services. So it's probably not the gem issue but then I wonder what pastebin service uses the correct font so I could share the unicode_plot output looking well.

Yeah, helping a friend of mine who is a biologist. They are doing it all in Python...

**Comment by kojix2** [comment]
I looked into the unicode_plot code.
Braille Pattern Blank is here.
https://github.com/red-data-tools/unicode_plot.rb/blob/caa4c5e055c44c2e48c317656ed0096f9a54b566/lib/unicode_plot/canvas/braille_canvas.rb#L27

This will be assigned to fill_char.
https://github.com/red-data-tools/unicode_plot.rb/blob/caa4c5e055c44c2e48c317656ed0096f9a54b566/lib/unicode_plot/canvas.rb#L17

I tried replacing Braille Pattern Blank with a normal space, but that doesn't seem to work. It shows up like this.

![image](https://user-images.githubusercontent.com/5798442/140238973-37cdc4c2-ee49-412c-8e91-32115db060f1.png)

@Nakilon 
Thanks for the reply. Actually, this is what it looks like on my display.

![image](https://user-images.githubusercontent.com/5798442/140238708-55866147-af9e-40c6-87b7-b12a13fa73c0.png)

> Yeah, helping a friend of mine who is a biologist. They are doing it all in Python...

Great!

**Comment by Nakilon** [comment]
Wow, that's unexpected. It's also without dots in Firefox and Safari. The only custom thing in my Chrome is zoom factor but changing it has no effect, hm.

UPD: the only related screenshot so far, but not helpful: https://stackoverflow.com/a/53137764/322020

**Comment by mrkn** [comment]
@Nakilon Is your computer Mac?

I can see this issue.  I use Chrome on MacBook Pro.

![image](https://user-images.githubusercontent.com/3959/150349102-e319f03b-f6a3-4a50-a815-4e9cafcefa20.png)

**Comment by mrkn** [comment]
I investigated what was going on in my environment, then I found some information.

First, the Braille glyphs were rendered with Apple Braille font.

![image](https://user-images.githubusercontent.com/3959/150351372-bef11d3e-98bf-4844-91d2-4f14719fc22c.png)

Second, Apple Braille font has five variations of Braille patterns:

![image](https://user-images.githubusercontent.com/3959/150351724-eaf6d647-7a27-4fad-a503-f0d290caabf9.png)

In this screenshot of the Font Book window, the variations are implemented as the font style, such as bold and italic.  I suspected that the Braille variation can be changed by specifying `font-style` or `font-weight` CSS property, and then I performed a simple experiment:

![image](https://user-images.githubusercontent.com/3959/150352853-5c4cf5c7-e970-426c-a0cb-003c2012c9b4.png)

As a result, I got the following:

![image](https://user-images.githubusercontent.com/3959/150352707-6d052d88-8635-4fb0-9e44-b6160aee2820.png)

But, I don't know why the Regular style is not selected.
